src/Interpreters/Context.cpp

```diff
@@ -2306,6 +2308,29 @@ void Context::clearUncompressedCache() const
         shared->uncompressed_cache->clear();
 }
 
+void Context::setPageCache(size_t bytes_per_chunk, size_t bytes_per_mmap, size_t bytes_total, bool use_madv_free, bool use_huge_pages)
+{
+    auto lock = getLock();
+
+    if (shared->page_cache)
+        throw Exception(ErrorCodes::LOGICAL_ERROR, "Page cache has been already created.");
+
+    shared->page_cache = std::make_shared<PageCache>(bytes_per_chunk, bytes_per_mmap, bytes_total, use_madv_free, use_huge_pages);
+}
+
+PageCachePtr Context::getPageCache() const
+{
+    auto lock = getLock();
+    return shared->page_cache;
+}
+
+void Context::dropPageCache() const
+{
+    auto lock = getLock();
+    if (shared->page_cache)
+        shared->page_cache->dropCache();
```

Comment:
is it ok to drop a page cache under a context lock? may be it could be done without it just under a page cache lock?